spec.md

```diff
@@ -515,15 +517,77 @@ Example:
 
 ## Registers
 
-TODO
+Registers are the stateful elements of a design.
 
-## Submodule Instances
+Registers can be connected (see [@sec:connects]).
+The state of the register is controlled through what is connected to it.
+
+The state of a register may be any passive type (see [@sec:passive-types]) and may not be `const`{.firrtl}.
+(TODO: is the `const` restriction a consequence of the passive part or not? Why is this required?)
```

Comment:
Is the `const` restriction a consequence of the passive part or not? Why is this required?

Intuitively, I get why it can't be `const`. Maybe it's just an additional restriction. But it might be important to motivate.

Reply:
@darthscsi explained that it's possible to get a time-varying `reg` in a circuit that only has `const` types.